src/IrcConnection/IrcConnection.cs

```diff
@@ -1079,61 +1127,20 @@ public ReadThread(IrcConnection connection)
             {
                 _Connection = connection;
                 QueuedEvent = new AutoResetEvent(false);
+                ThreadName = "ReadThread ("+_Connection.Address+":"+_Connection.Port+")";
             }
 
-            /// <summary>
-            /// 
-            /// </summary>
-            public void Start()
-            {
-                _Thread = new Thread(new ThreadStart(_Worker));
-                _Thread.Name = "ReadThread ("+_Connection.Address+":"+_Connection.Port+")";
-                _Thread.IsBackground = true;
-                _Thread.Start();
-            }
+            protected override void PrepareStart() { }
 
-            /// <summary>
-            /// 
-            /// </summary>
-            public void Stop()
+            protected override void Worker()
             {
 #if LOG4NET
-                _Logger.Debug("Stop()");
-#endif
-                
-#if LOG4NET
-                _Logger.Debug("Stop(): aborting thread...");
-#endif
-                _Thread.Abort();
-                // make sure we close the stream after the thread is gone, else
-                // the thread will think the connection is broken!
-#if LOG4NET
-                _Logger.Debug("Stop(): joining thread...");
-#endif
-                _Thread.Join();
-                
-#if LOG4NET
-                _Logger.Debug("Stop(): closing reader...");
-#endif
-                try {
-                    _Connection._Reader.Close();
-                } catch (ObjectDisposedException) {
-                }
-
-                // clean up our receive queue else we continue processing old
-                // messages when the read thread is restarted!
-                _Queue.Clear();
-            }
-
-            private void _Worker()
-            {
-#if LOG4NET
-                Logger.Socket.Debug("ReadThread started");
+                Logger.Socket.Debug("ReadThread Worker(): starting");
 #endif
                 try {
                     string data = "";
                     try {
-                        while (_Connection.IsConnected &&
+                        while (!IsStopRequested && _Connection.IsConnected &&
                                ((data = _Connection._Reader.ReadLine()) != null)) {
```

Comment:
IsStopRequested should be checked separately
reading from the connection should have a timeout set so that we can stop without receiving a message that we will potentially just drop because we have already stopped